XCUITests/BaseTestCase.swift

```diff
@@ -7,21 +7,32 @@ import XCTest
 
 class BaseTestCase: XCTestCase {
     var navigator: MMNavigator<FxUserState>!
-    var app: XCUIApplication!
+    let app =  XCUIApplication()
     var userState: FxUserState!
 
+    // These are used during setUp(). Change them prior to setUp() for the app to launch with different args,
+    // or, use restart() to re-launch with custom args.
+    var launchArguments = [LaunchArguments.ClearProfile, LaunchArguments.SkipIntro, LaunchArguments.SkipWhatsNew]
+
+    func setUpScreenGraph() {
+        navigator = createScreenGraph(for: self, with: app).navigator()
+        userState = navigator.userState
+    }
+
+    func setUpApp() {
+        app.launchArguments = [LaunchArguments.Test] + launchArguments
+        app.launch()
+    }
+
     override func setUp() {
         super.setUp()
         continueAfterFailure = false
-        app = XCUIApplication()
-        app.terminate()
-        restart(app, args: [LaunchArguments.ClearProfile, LaunchArguments.SkipIntro, LaunchArguments.SkipWhatsNew])
-        navigator = createScreenGraph(for: self, with: app).navigator()
-        userState = navigator.userState
+        setUpApp()
+        setUpScreenGraph()
```

Comment:
Made convenience functions so that if someone needs to overrride setUp() and customize it without calling super.setUp(), they can easily call setUpApp() and/or setUpScreenGraph() as-needed in their overrride.

Reply:
This looks good! Will be helpful in the future